packages/core/src/credential/Credential.ts

```diff
@@ -292,118 +277,114 @@ export function fromClaim(
   return credential
 }
 
-export interface TrustPolicy {
-  /**
-   * If set to true, attestations issued directly by this identity are accepted.
-   */
-  isAttester?: boolean
-  /**
-   * If set to true, attestations linked to a delegation node which this identity has control over are accepted.
-   */
-  isDelegator?: boolean
+type VerifyOptions = {
+  ctype?: ICType
+  challenge?: string
+  didResolveKey?: DidResolveKey
+  throwOnRevoked?: boolean
+}
+
+/**
+ * Verifies data structure & data integrity of a credential object.
+ * This combines all offline sanity checks that can be performed on an ICredential object.
+ * A credential is valid only iff it is well formed and there is an onchain attestation record referencing its root hash.
```

Comment:
```suggestion
 * A credential is valid only if it is well formed and there is an on-chain attestation record referencing its root hash.
```

Reply:
I’m not the only one thrown off by the usage of "iff" in the sense "if and only if" :)

Reply:
Is this even true? If I have a well structured credential with an on-chain attestation record it can still be invalid? E.g. if the delegation record is wrong or doesn't exist.

Reply:
You're right, this isn't complete; the attestation also needs to match the credential, i.e. delegation and ctype ids need to be identical to what was committed to in the credential. BTW that's something we should have designed differently; the runtime should combine at least delegation & ctype info into the credential hash so there is no chance these data do not match if the hash is identical. Ideally also the attester and block number, just to make sure the same attestation is never re-created.

Reply:
Anyway, we would do these checks as part of verifyCredential now.

Reply:
This should be part of the discussion for credentials v2. We can build on the current limitations and do better.